tests/js/chartAuthoringTests.js

```diff
@@ -155,8 +160,50 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         jqUnit.assertDeepEq("The reset has an aria-controls attribute properly referencing the form, pie, legend and total", resetAriaControlsAttr, dataEntryFormlId + " " + legendTableId + " " + pieChartPieId + " " + dataEntryFormTotalId);
     };
 
+    floe.tests.chartAuthoringTester.verifyUpdate = function (that) {
+        floe.tests.chartAuthoringTester.verifyRelay(that);
+        floe.tests.chartAuthoringTester.verifyUpdateDataEntryPanel(that);
+    };
+
+    // Verify that the updateDataEntryPanel function updates the UI
+    floe.tests.chartAuthoringTester.verifyUpdateDataEntryPanel = function (that) {
+        // floe.tests.chartAuthoring.updateDataSet
+        var dataEntries = that.chartAuthoringInterface.dataEntryPanel.locate("dataEntry");
+
+        var dataEntryLabelSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.label;
+        var dataEntryValueSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.value;
+
+        dataEntries.each(function(idx) {
+            var expectedData = floe.tests.chartAuthoring.updateDataSet[idx];
+
+            if(expectedData !== undefined) {
```

Comment:
@cindyli It can be undefined because the length of expectedData may be less than the length of dataEntries (which is always 5). That said, I don't think this is quite the right approach to the test and would miss certain cases - I'll refactor.